### PR TITLE
Use embedded scaledobject templates

### LINKED ISSUE #72: Use kedacore/keda ScaledObject types
Instead of using `unstructured.Unstructured` in the functionality where we [create `ScaledObject`s](https://github.com/kedacore/http-add-on/blob/main/pkg/k8s/scaledobject.go#L30), we should use the actual [`ScaledObject` type](https://pkg.go.dev/github.com/kedacore/keda/v2@v2.1.0/api/v1alpha1#ScaledObject) in KEDA core itself. From experience, we'll probably need to do #64 first.

### PR DESCRIPTION
**Pull request by khaosdoctor**
Removing the unstructured code object in the scaled object creation and started using go embedded templates as per what @coderanger suggested.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] A PR is opened to update the documentation on [our docs repo](https://github.com/kedacore/keda-docs)

Fixes #72

### CODE REVIEW

**Comment by khaosdoctor** [comment]
@arschles can you please give it a review?

**Comment by coderanger** [comment]
Nifty, much simpler than my old thing using vfsgen.

**Comment by zroubalik** [comment]
I am curious, is there a specific reason, why don't you want to import the type from keda? You don't want to mess with dependencies? Or what is the benefit?

**Comment by coderanger** [comment]
I explained it a bit on Slack but the main reason is you cannot use concrete structs with Server Side Apply and if this controller isn't using that already, it should soon. So if you have to use an Unstructured anyway, might as well make it pretty. As a secondary issue, yes dependency hell is nice to avoid.

That first issue might some day go away when controller-gen grows support for the typed apply builder syntax but that's a long way off right now (core support is going in to client-go for 1.21 as a "no really we mean it" alpha).

**Comment by zroubalik** [comment]
Thanks, that make sense.

> That first issue might some day go away when controller-gen grows support for the typed apply builder syntax but that's a long way off right now (core support is going in to client-go for 1.21 as a "no really we mean it" alpha).

Is this being tracked in some kubebuilder (controller-runtime) issue? I wasn't able to find anything.

**Comment by coderanger** [comment]
Not yet, not that far along :) The core support in client-go was only merged a few days ago.

**Comment by arschles** [comment]
>You don't want to mess with dependencies?

@zroubalik this certainly is a nice thing, though

File: pkg/k8s/scaledobject.go [hunk 1]
```diff
@@ -52,33 +61,30 @@ func NewScaledObject(
...
 		labels[k] = vIface
 	}
+
+	tpl, err := template.ParseFS(scaledObjectTemplateFS, "templates/scaledobject.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var scaledObjectTemplateBuffer bytes.Buffer
+	tpl.Execute(&scaledObjectTemplateBuffer, map[string]interface{}{
```
Comment (arschles) [inline pkg/k8s/scaledobject.go:71]: @khaosdoctor can you check the error here please?

File: pkg/k8s/scaledobject.go [hunk 2]
```diff
@@ -52,33 +61,30 @@ func NewScaledObject(
...
+		"MinReplicas": minReplicas,
+		"MaxReplicas": maxReplicas,
+		"DeploymentName": deploymentName,
+		"ScalerAddress": scalerAddress,
+	})
+
+	var decodedYaml map[string]interface{}
+	decodeErr := yaml.Unmarshal(scaledObjectTemplateBuffer.Bytes(), &decodedYaml)
+	if decodeErr != nil {
+		log.Fatal(err)
```
Comment (arschles) [inline pkg/k8s/scaledobject.go:84]: @khaosdoctor instead of `log.Fatal`, can you please return the error?

**Review by arschles — CHANGES_REQUESTED** [review]
@khaosdoctor a few requests

**Comment by khaosdoctor** [comment]
Will do :)!

**Comment by khaosdoctor** [comment]
@arschles done!

**Review by arschles — APPROVED** [review]
looks great @khaosdoctor ✌️ 🚀

**Pull request merged**